File: archived/src/python/ch01/1.3/l1/exercises_starter.py
from gcd_lcm import gcd, lcm
from modular_identities import mod_add, mod_mul
from fractions_utils import reduce_fraction

def exercise_1():
    # gcd(252, 198)
    return gcd(252, 198)

def exercise_2():
    # lcm(12, 18)
    return lcm(12, 18)

def exercise_3():
    # (37+85)%12 equals ((37%12)+(85%12))%12
    lhs = (37 + 85) % 12
    rhs = mod_add(37, 85, 12)
    return lhs, rhs

def exercise_4():
    # reduce 84/126
    return reduce_fraction(84, 126)

def exercise_5():
    # smallest day multiple of 12 and 18 (i.e., LCM)
    return lcm(12, 18)

def main():
    print("Exercise 1 gcd(252,198):", exercise_1())
    print("Exercise 2 lcm(12,18):", exercise_2())
    print("Exercise 3 modular add check:", exercise_3())
    print("Exercise 4 reduce 84/126:", exercise_4())
    print("Exercise 5 smallest day multiple:", exercise_5())

if __name__ == "__main__":
    main()

File: archived/src/c/ch01/1.2/l0/exercises_starter.c
#include <stdio.h>

int exercise_1(void) { return 326 + 589; }
int exercise_2(void) { return 704 - 259; }
int exercise_3(void) { return 38 * 12; }
void exercise_4(int *q, int *r) { *q = 123 / 7; *r = 123 % 7; }

int main(void) {
    printf("Exercise 1 (326+589): %d\n", exercise_1());
    printf("Exercise 2 (704-259): %d\n", exercise_2());
    printf("Exercise 3 (38*12): %d\n", exercise_3());
    int q, r;
    exercise_4(&q, &r);
    printf("Exercise 4 (123//7, 123%%7): %d %d\n", q, r);
    return 0;
}
